exo-create/main_test.go

```diff
@@ -0,0 +1,162 @@
+package main_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/DATA-DOG/godog"
+	"github.com/DATA-DOG/godog/gherkin"
+	"github.com/Originate/exosphere/exo-create/test_helpers"
+)
+
+var cmd *exec.Cmd
+var in io.WriteCloser
+var out bytes.Buffer
+var appDir string
+var err error
+
+type AppConfig struct {
+	AppName, AppVersion, ExocomVersion, AppDescription string
+}
+
+func run(cwd, command string) error {
+	cmdSegments := strings.Split(path.Join(cwd, "bin", command), " ")
+	cmd = exec.Command(cmdSegments[0], cmdSegments[1:]...)
+	cmd.Dir = appDir
+	in, err = cmd.StdinPipe()
+	cmd.Stdout = &out
+	if err != nil {
+		return err
+	}
+	if err = cmd.Start(); err != nil {
+		return fmt.Errorf("Error running %s\nError:%s", command, err)
+	}
+	return nil
+}
+
+func validateTextContains(haystack, needle string) error {
+	if strings.Contains(haystack, needle) {
+		return nil
+	}
+	return fmt.Errorf("Expected:\n\n%s\n\nto include\n\n%s", haystack, needle)
+}
+
+func enterInput(row *gherkin.TableRow) error {
+	_, input := row.Cells[0].Value, row.Cells[1].Value
+	_, err := in.Write([]byte(input + "\n"))
+	return err
+}
+
+// nolint gocyclo
+func FeatureContext(s *godog.Suite) {
+	var cwd, childOutput string
+	var err error
+
+	s.BeforeSuite(func() {
+		var err error
+		cwd, err = os.Getwd()
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	s.Step(`^executing "([^"]*)"$`, func(command string) error {
+		return run(cwd, command)
+	})
+
+	s.Step(`^starting "([^"]*)" in the terminal$`, func(command string) error {
+		appDir = path.Join(cwd, "tmp")
+		if err := testHelpers.EmptyDir(appDir); err != nil {
+			return err
+		}
+		return run(cwd, command)
+	})
+
+	s.Step(`^entering into the wizard:$`, func(table *gherkin.DataTable) error {
+		for _, row := range table.Rows[1:] {
+			if err := enterInput(row); err != nil {
+				return err
+			}
+		}
+		defer in.Close()
+		return nil
+	})
+
+	s.Step(`^running "([^"]*)" in the terminal$`, func(command string) error {
+		appDir = path.Join(cwd, "tmp")
+		if err := testHelpers.EmptyDir(appDir); err != nil {
+			return err
+		}
+		return run(cwd, command)
+	})
+
+	s.Step(`^waiting until I see "([^"]*)" in the terminal$`, func(expectedText string) error {
+		if err := cmd.Wait(); err != nil {
+			return err
+		}
+		childOutput = out.String()
+		return validateTextContains(childOutput, expectedText)
```

Comment:
This step should keep calling `cmd.Wait` until it either finds the given string in the output, or hits a timeout. Please read https://blog.golang.org/go-concurrency-patterns-timing-out-and for some background.